single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/AbstractLabelProvider.java

```diff
@@ -105,9 +105,7 @@ protected LabelPosition getLabelPosition(Node node, Direction direction) {
         String positionName = "";
         double angle = 0;
         if (direction != UNDEFINED) {
-            yShift = direction == TOP
-                    ? -LABEL_OFFSET
-                    : ((int) (componentLibrary.getSize(node.getComponentType()).getHeight()) + LABEL_OFFSET);
+            yShift = (direction == TOP ? -layoutParameters.getMaxComponentHeight() : layoutParameters.getMaxComponentHeight()) + (componentLibrary.getSize(node.getComponentType()).getHeight() / 2);
```

Comment:
Now the shift you added depends on maxComponentHeight, that is if the maxComponentHeight is 100px you have the label 50px above the biggest component. You need to divide maxComponentHeight by 2 to have a "fixed" shift. And probably add an extra offset to about that the label is not touching the biggest component.